DigitalLearningSolutions.Data/Helpers/LearningHubSsoSecurityHelper.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using DigitalLearningSolutions.Data.Services;
+    using Microsoft.Extensions.Configuration;
+
+    public interface ILearningHubSsoSecurityHelper
+    {
+        string GenerateHash(string state, string secretKey);
+
+        bool VerifyHash(string state, string secretKey, string hash);
+    }
+
+    public class LearningHubSsoSecurityHelper : ILearningHubSsoSecurityHelper
+    {
+        private static readonly DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        private readonly IClockService clockService;
+        private readonly IConfiguration config;
+
+        public LearningHubSsoSecurityHelper(IClockService clockService, IConfiguration config)
+        {
+            this.clockService = clockService;
+            this.config = config;
+        }
+
+        public string GenerateHash(string state, string secretKey)
+        {
+            var secondsSinceEpoch = GetSecondsSinceEpoch();
+            var encoder = new UTF8Encoding();
+            var salt = encoder.GetBytes(secretKey);
+            var timedState = encoder.GetBytes(state + secondsSinceEpoch);
+            return GetHash(timedState, salt);
+        }
+
+        public bool VerifyHash(string state, string secretKey, string hash)
+        {
+            var secondsSinceEpoch = GetSecondsSinceEpoch();
+            var encoder = new UTF8Encoding();
+            var salt = encoder.GetBytes(secretKey);
+            var toleranceInSec = config.GetSection("LearningHubSSO").GetValue<int>("ToleranceInSeconds");

```

Comment:
Although we are probably only using these config values here we should stick to the convention that is in place.
If we had the fullness of time I wuld actually opt for the `IOptions` approach for a strongly typed and easily `mocked` implementation over the standard `IConfiguration` approach but I think that is going to have to wait.
In the mean time can we look to implement some extension methods in the ConfigHelper and use those to match other areas like the `ResourcesController` ?